packages/btcindexer/src/btcindexer.ts

```diff
@@ -683,3 +725,40 @@ function parseSuiRecipientFromOpReturn(script: Buffer): string | null {
 	// for now we cannot determine the recipient
 	return null;
 }
+
+async function getSenderInsertStmts(
+	tx: Transaction,
+	d1: D1Database,
+	electrsUrl: string,
+): Promise<D1PreparedStatement[]> {
+	const senderAddresses = new Set<string>();
+	const insertStmt = d1.prepare(
+		"INSERT OR IGNORE INTO nbtc_deposit_senders (tx_id, sender) VALUES (?, ?)",
+	);
+
+	const prevTxFetches = tx.ins.map(async (input) => {
+		const prevTxId = Buffer.from(input.hash).reverse().toString("hex");
+		const prevTxVout = input.index;
+
+		try {
+			const response = await fetch(`${electrsUrl}/tx/${prevTxId}`);
+			if (!response.ok) return;
+
+			const prevTx = (await response.json()) as { vout: { scriptpubkey_address?: string }[] };
+			const prevOutput = prevTx.vout[prevTxVout];
+			if (prevOutput?.scriptpubkey_address) {
+				senderAddresses.add(prevOutput.scriptpubkey_address);
+			}
```

Comment:
what if there is no `scriptpubkey_address` ?

Reply:
we just skip it

Reply:
i mean, we index it just without the sender address, i dont think there is much we can do about it